docs/rules/no-array-reduce.md

```diff
@@ -46,3 +46,21 @@ for (const element of array) {
 	result += element;
 }
 ```
+## Options
+
+### allowNumericInitialValue
+
+Type: `boolean`\
+Default: `false`
+
+Can be enabled to allow `reduce` to be used to sum numbers.
```

Comment:
I think the description should first describe the option and then afterwards what it enables (the above).